### PR TITLE
DM-33965: Add PSF ellipticity quiver plots

### PR DESCRIPTION
**Pull request by arunkannawadi**

### CODE REVIEW

File: pipelines/visitQAEllipPlots.yaml [hunk 1]
```diff
@@ -201,6 +201,31 @@ tasks:
...
+      axisActions.zAction.colB.colXy: "ixyPSF"
+      connections.plotName: Ediff_ixxPSF_quiver
+      selectorActions.flagSelector: VisitPlotFlagSelector
+      selectorActions.SnSelector: SnSelector
+      selectorActions.SnSelector.bands: [""]
+      selectorActions.SnSelector.fluxType: "psfFlux"
+      selectorActions.SnSelector.threshold: 10
+      axisLabels: {"x": "R.A. (Degrees)", "y": "Dec. (Degrees)", "z": "E Difference (eMEAS - ePSF)"}
+      python: |
+        from lsst.analysis.drp.dataSelectors import VisitPlotFlagSelector, GalaxyIdentifier, SnSelector
```
Comment (cmsaunders) [inline pipelines/visitQAEllipPlots.yaml:225]: You don't use `GalaxyIdentifier`--do you mean to be selecting only galaxies?

**Review by cmsaunders — APPROVED** [review]
I don't think there should be any calculations in the `skyplot` function, but I am okay pushing that off to another ticket. Otherwise, I have just requested some minor changes. I would like to see a more informative key on the plots, but I hope that can be done quickly.

File: pipelines/visitQAEllipPlots.yaml (hunk 1 above)
Comment (arunkannawadi) [reply pipelines/visitQAEllipPlots.yaml:225]: I should remove it. I never need to select on galaxies to make these plots.

File: python/lsst/analysis/drp/quiverPlotVisit.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+    visitSummaryTable = pipeBase.connectionTypes.Input(doc="A summary table of the ccds in the visit",
+                                                       storageClass="ExposureCatalog",
+                                                       name="visitSummary",
+                                                       dimensions=("visit",))
+
+    quiverPlot = pipeBase.connectionTypes.Output(doc="A plot showing the on-sky distribution of a value.",
+                                                 storageClass="Plot",
+                                                 name="quiverPlotVisit_{plotName}",
+                                                 dimensions=("visit",))
```
[the remark this answers is not in the document]
Comment (arunkannawadi) [reply python/lsst/analysis/drp/quiverPlotVisit.py:29]: `skyPlotVisit` doesn't have `skymap` as one of its dimensions, so I decided to stick with only "visit" as was the case there.

File: python/lsst/analysis/drp/quiverPlot.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+                    ax.annotate(dataId, (cenX, cenY), color="k", fontsize=7, ha="center", va="center",
+                                path_effects=[pathEffects.withStroke(linewidth=2, foreground="w")])
+
+        for (i, (xs, ys, colorVals, cmap, label)) in enumerate(toPlotList):
+            if "tract" not in sumStats.keys() or not self.config.plotOutlines:
+                minRa = np.min(xs)
+                maxRa = np.max(xs)
+                minDec = np.min(ys)
+                maxDec = np.max(ys)
+            e = np.hypot([_e.real for _e in colorVals], [_e.imag for _e in colorVals])
```
[the remark this answers is not in the document]
Comment (arunkannawadi) [reply python/lsst/analysis/drp/quiverPlot.py:217]: That's true, and completely agree with there being no calculation inside the `skyPlot` routine. This came about in trying to use a concrete `SkyPlotTask` as a base class to reuse its `runQuantum` and `run` methods. I'll mark it as a TODO in DM-34057.

File: python/lsst/analysis/drp/quiverPlot.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+            # Ellipiticity is |e|*exp(i*2*theta), but we want to plot
+            # |e|*exp(i*theta). So we multiply by |e| and take square root
+            # instead of more expensive trig calls.
+            e_scaled = np.sqrt(e*colorVals)
+            c1, c2 = np.real(e_scaled), np.imag(e_scaled)
+            quiverOut = ax.quiver(xs, ys, c1, c2, e, angles="uv", scale=None, pivot="middle",
+                                  units="width", width=0.001, headwidth=0.0, headlength=0.0,
+                                  headaxislength=0.0, label=label, cmap=cmap)
+            # Should the quiverKey go into plotInfo?
+            ax.quiverkey(quiverOut, 0.1, 0.1, 1,
```
[the remark this answers is not in the document]
Comment (arunkannawadi) [reply python/lsst/analysis/drp/quiverPlot.py:227]: Would it help to have two keys, one horizontal and one at 45 degrees showing what ellipticities they correspond to? Also, I highly doubt that the scale in the key is the same as the scale in the plot, so I am inclined to drop it altogether at this point.

[the remark this answers is not in the document]
Comment (cmsaunders) [reply python/lsst/analysis/drp/quiverPlot.py:227]: I think what you are describing would work.
The scale in the existing key does seem plausible to me, comparing its length to the big outliers in the plot.

[the remark this answers is not in the document]
Comment (arunkannawadi) [reply python/lsst/analysis/drp/quiverPlot.py]: Oh, I made hacked it to show a whisker of length 1 although the key says 0.142. The key seemed to be out of scale. I have made it a config option that defaults to not showing the key. The formatting is punted to DM-34057.

File: python/lsst/analysis/drp/quiverPlot.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (arunkannawadi) [reply python/lsst/analysis/drp/quiverPlot.py]: I didn't want to accrue too much technical deblt, so this is taken care of in the current version. The calculations are done in the `CalcEDiff` action and the `skyPlot` within `quiverPlot` simply expects the quantity to be plotted as a complex value.

**Pull request merged**